web-api/src/mind.rs

```diff
@@ -223,10 +224,15 @@ where
 }
 
 /// Runs the user-based mind benchmark
-fn run_benchmark() -> Result<(), Error> {
+async fn run_user_benchmark() -> Result<(), Error> {
     let document_provider = DocumentProvider::new("news.tsv")?;
+    let impressions: DeserializeRecordsIntoIter<File, _> = read("behaviors.tsv")?;
```

Comment:
:art: the type hint is not required
```suggestion
    let impressions = read("behaviors.tsv")?;
```

also, we could just call the iterator construction in the loop head where it's actually used, in that way the iterator construction is not visually detached from the iterator usage (in l.242):
```rust
    for impression in read("behaviors.tsv")? {
```

Reply:
makes sense